### DISCUSSION TITLE
Test local themes

### DISCUSSION THREAD

**Discussion by rubiin** [kRHYME7/Hyde-cli#87] [Q&A] [answered]
How to test local theme. I made a certain theme and want to test it without adding it to github

**Comment by kRHYME7** [comment]
![image](https://github.com/user-attachments/assets/07224330-b001-4b44-8c10-1ef4b9dadd95)

The patcher is actually made for local use and was extended to work with github. 
![image](https://github.com/user-attachments/assets/0438851d-f381-4060-9e48-b0e05055eaa4)


For testing Locally theme should be inside a `  ../Configs ` directory with this tree 

![image](https://github.com/user-attachments/assets/416067bd-3c50-4070-b3c4-bb9cffc3a18d)
Configs are typically the same as how HyDE stores the configs.  
` Configs/.config `  ==> ` $HOME/.config ` 

Source will have the Tarball for Gtk theme, gtk cursor, gtk icon , etc.
![image](https://github.com/user-attachments/assets/e2401a42-c4a2-45f6-b949-ca7e559bdee4)

![image](https://github.com/user-attachments/assets/173b62cb-13f4-48a5-b85d-5e1f98e8eaa2)

Note that to avoid errors the parent directory for the contents of the tarball should be the same of what is declared in `../Configs/.config/hypde/themes/THEME-NAME/hypr.theme` 
![image](https://github.com/user-attachments/assets/bc47d16d-c27b-4437-b682-925598ed022b)
Above is the old implementation, patcher now supports using hyprland variables like ` $GTK-THEME `  instead of parsing from a gsettings command. (This might not work if uses scripts is not updated )


Note Normally, when you reimport (re patch) a theme it will skip the extraction of the tarball, so you need this flags if you want to populate the changes from the tarball
![image](https://github.com/user-attachments/assets/401b1ced-b077-40e8-9e6e-2c58c8f0d89d)
 

Pretty much that's it.

**Reply by kRHYME7** [reply, marked as answer]
If Users just want to make a theme for themselves then above method is  irrelevant. 

Just copy a theme directory inside  ` ~/.config/hyde/themes ` then choose that theme  
![image](https://github.com/user-attachments/assets/f9bafff4-3375-4b24-9e3c-c5605a51f02f)
![image](https://github.com/user-attachments/assets/ff4100b7-8d89-40a9-b370-e210a829598a)

Then users can do theme changes on that copied directory

**Reply by rubiin** [reply]
Also when making a new theme, do we need to edit kvantum files

**Reply by kRHYME7** [reply]
edit the generated one or find some kvantum theme rename it the same as the other themes. (do not forget to add the header line )

you need the 2 files the svg file and the kconfig file.
